cmake/everest-generate.cmake

```diff
@@ -494,7 +507,102 @@ function (ev_add_py_module MODULE_NAME)
             add_subdirectory(${MODULE_PATH})
         endif()
     else()
-        message(WARNING "C++ module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        message(WARNING "Python module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        return()
+    endif()
+
+    # FIXME (aw): this will override EVEREST_MODULES, might not what we want
+    set_property(
+        GLOBAL
+        PROPERTY EVEREST_MODULES ${EVEREST_MODULES}
+    )
+endfunction()
+
+function (create_rust_cargo_toml RUST_MODULE_LIST)
+    if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+        return()
+    endif()
+
+    set(OUTPUT_FILE_PATH "${CMAKE_BINARY_DIR}/modules/Cargo.toml")
+
+    # Write the static parts
+    file(WRITE "${OUTPUT_FILE_PATH}" "[workspace]\nresolver = \"2\"\nmembers = [\n")
+
+    # Append each symlink path
+    foreach(SYMLINK ${RUST_MODULE_LIST})
+        file(APPEND "${OUTPUT_FILE_PATH}" "    \"${SYMLINK}\",\n")
```

Comment:
cant we use just the full path here, instead of symlinking?

Reply:
no, every member of a Cargo workspace must be in a subdirectory of a cargo workspace, hence symlinking is the only thing that works in the general case.